### ISSUE TITLE
Recommended fix to accommodate hamlib change

### ISSUE THREAD

**Issue by markjfine** [fillods/grig#22]
Hamlib made a change to the way rig_list_add is referenced. This diff basically removes `const` in the prototype and calling structure from grig_list_add.

[hamlib_change.diff.txt](https://github.com/fillods/grig/files/15055590/hamlib_change.diff.txt)

**Comment by srs4511351** [comment]
I believe these changes are for hamlib 4.6 which is now the current release.
After implementing the recommended changes, I still have make errors.
```
rig-selector.c:49:13: error: conflicting types for ‘connect’; have ‘void(GtkWidget *, void *)’ {aka ‘void(struct _GtkWidget *, void *)’}
   49 | static void connect (GtkWidget *, gpointer);
      |             ^~~~~~~
In file included from /usr/local/include/hamlib/rig.h:50,
                 from rig-selector.c:35:
/usr/include/aarch64-linux-gnu/sys/socket.h:126:12: note: previous declaration of ‘connect’ with type ‘int(int,  const struct sockaddr *, socklen_t)’ {aka ‘int(int,  const struct sockaddr *, unsigned int)’}
  126 | extern int connect (int __fd, __CONST_SOCKADDR_ARG __addr, socklen_t __len);
      |            ^~~~~~~
```
Here is the configure and make log
[grig build log hamlib 4.6 with fixes.txt](https://github.com/user-attachments/files/18255685/grig.build.log.hamlib.4.6.with.fixes.txt)

**Comment by markjfine** [comment]
My version has `connect()` changed to `connectrig()` in `src/rig-selector.c` in my local git version on lines 49, 194, and 442. I may have done that to get around the conflict with a separate prototype of connect() that you're seeing.

I likely meant to post something, got distracted, and then forgot about it. I certainly don't see that change in the github repository, nor do I see an issue posted. Maybe I should just do a complete diff of what I have against github and repost. What I have locally builds fine and runs.

Edit: Turns out those three lines in rig-selector.c are the only other changes I have from the repository.
[hamlib_change2.diff.txt](https://github.com/user-attachments/files/18256324/hamlib_change2.diff.txt)

**Comment by srs4511351** [comment]
Thanks! That worked.
I used version 0.9.1~git which is a line or two off from your diff file, but works if you match up the lines.

I hope the maintainers use this. It needs to be able to work for hamlib 4.5 and 4.6.

**Comment by markjfine** [comment]
I just clone it from the repository. I have a script that I periodically run that does a round-robin `git pull origin main` (or master) on about 10 different repositories. Then I just rebuild what needs rebuilding. If there's a hamlib change, and invariably there is, I always rebuild grig right afterwards.

**Comment by markjfine** [comment]
Lol... And hamlib just put `const` back in the `rig_caps` declaration, nullifying the first fix above.

**Comment by srs4511351** [comment]
Cool. Now we have to back through and unfix everything!

**Comment by markjfine** [comment]
Keep in mind that was done in the repository after 4.6 was released. If you only pull official releases, you won't notice a change.

**Comment by srs4511351** [comment]
It seems that the master branch has the changes, as software that failed on
hamlib is now compiling successfully.

----Steve

**Comment by srs4511351** [comment]
There is still a problem compiling grig with hamlib 4.7.
grig's configure script reports:
```
grig version....... : 0.9.1~git
Hamlib version..... : 4.7~git
```

Hamlib 4.7 has reverted rig_caps to const, but it is still failing on "connect".
This works on all of the other applications that expect Hamlib 4.5.5.
I had to change "connect" to "connectrig" to get grig to compile. I did not have to remove "const" from the rig_caps lines.
Is there something else that Hamlib changed in 4.6+ ?

**Comment by markjfine** [comment]
That is correct. Only `const` was to be reinserted back into `rig_caps`. The other change from `connect()` to `connectrig()` was still needed and should not have been reverted.
[hamlib_change3.diff.txt](https://github.com/user-attachments/files/18298885/hamlib_change3.diff.txt)

**Comment by thadigus** [comment]
Bit of a newbie here; could someone quickly detail the changes to get this project to compile as of right now? I'm running into the same issue but I think I'm lost in the noise. Is there a PR or a fixed fork?

**Comment by markjfine** [comment]
All you have to do is edit src/rig-selector.c and change `connect` to `connectrig` in lines 49, 194, and 442.

**Comment by markjfine** [comment]
Just in case, I've forked this at https://github.com/markjfine/grig. It has the necessary changes and will be looking at squashing some of the warnings as well.

**Comment by srs4511351** [comment]
I tried your fork and it it works, thanks.

**Comment by DaveKBV** [comment]
Hi.

After wasting a lot of time going round in url circles, I found this set of github pages.

I ended up manually editing the source, and indeed it works now.  Thanks markJfine!
Using Hamlib 4.6.4 on LMDE6 (Debian 12 derived)

Who needs to know, so that the grig main site (as found by Google) references this set of pages for up to date info, rather than the somewhat out of date SF pages?

73   Dave G8KBV

**Comment by fillods** [comment]
The "Fixes clash with Hamlib" a1c33e30 by @markjfine just got cherry-picked. Thanks. Feel free to provide a PR next time.

NB: I let aside the "Update for gettext 0.22"  037046bb because Ubuntu 24.04.x only has 0.21 and the commit requires bumping to **0.23.1**. BTW Mark, why bumping to this version ?

**Comment by markjfine** [comment]
> The "Fixes clash with Hamlib" [a1c33e3](https://github.com/fillods/grig/commit/a1c33e300bd8231d021011b6b5d5bc9711448ebb) by [@markjfine](https://github.com/markjfine) just got cherry-picked. Thanks. Feel free to provide a PR next time.

Cheers. I should have done so and normally would, but didn't because I wasn't sure if you were still working this if I'm honest.

> NB: I let aside the "Update for gettext 0.22" [037046b](https://github.com/fillods/grig/commit/037046bb991ab27ca55e1e921d1155ad6ef9f819) because Ubuntu 24.04.x only has 0.21 and the commit requires bumping to **0.23.1**. BTW Mark, why bumping to this version ?

Likely because Homebrew updated it to 0.23.1 o/a Dec 2024 (when the fix was provided). Homebrew is what I use as a Linux-compatible package manager on macOS. It's quite possible that Fedora went to it around the same time as well, so I may have assumed it was universal.

As of today, Fedora 44 is currently using 0.26.5 and Homebrew is now using 1.0 of all things, but neither is affecting anything that I know of. I personally don't use Ubuntu, but I've found that it tends to be quite conservative, favouring complete interoperability before playing the 'latest and greatest' game.
